.github/workflows/core-test.yml

```diff
@@ -40,18 +44,39 @@ jobs:
         - name: Checkout repository
           uses: actions/checkout@v3
 
+        - name: Get changed files
+          id: changed-files
+          uses: tj-actions/changed-files@v35
+
+        - name: Module changes
+          id: modules-changed
+          # check if core (or toml) has been changed.
+          run: |
+            for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+              if [[ $file == *"/core/"* || $file == *"/pyproject.toml" ]]; then
+                echo "Core changed"
+                echo "CORE_CHANGED=true" >> $GITHUB_OUTPUT
+              fi
+            done
+
```

Comment:
I think you will need to use bash for this step otherwise it would fail in windows. 
So you can control the bash with  the `shell: bash` instruction:

https://github.com/SpikeInterface/spikeinterface/blob/91998fa7634a7fe75b8097940d3efc52271683a3/.github/workflows/core-test.yml#L45-L46